include/roboteam_proto/Publisher.h

```diff
@@ -7,17 +7,37 @@
 
 #include <string>
 #include <zmqpp/zmqpp.hpp>
+#include <google/protobuf/message.h>
+#include "Channel.h"
 
 namespace roboteam_proto {
+
+template <class T>
 class Publisher {
  private:
   zmqpp::context context;
   zmqpp::socket * socket;
-  std::string tcpPort;
+  Channel channel;
+
  public:
-  explicit Publisher(const std::string & tcpPort);
-  ~Publisher();
-  bool send(std::string const& topic, std::string message);
+  Publisher(const Publisher & copy) = delete; // you cannot copy this object
```

Comment:
We could consider starting to use the rule of three as a paradigm. Here it would be quite relevant for example, I recommend you also also delete the copy assignment operator. (Publisher::operator=(const Publisher & copy) = delete). Same story for Subsrciber.h

Reply:
Good point. I agree